src/encode.py

```diff
@@ -4,34 +4,41 @@
 TIME_WINDOWS_DEFAULT = ('60s', '1h', '1d', '5d', '30d', '365d')
 COUNTERS_DEFAULT = ('attempts', 'wins')
 
-def encode_df(df, features=FEATURES_DEFAULT, counters = COUNTERS_DEFAULT, time_windows=TIME_WINDOWS_DEFAULT):
+def encode_df(df: pd.DataFrame, features: tuple=FEATURES_DEFAULT, counters: tuple=COUNTERS_DEFAULT, time_windows: tuple=TIME_WINDOWS_DEFAULT):
     '''
     Get the wanted counters of each student and dummifies the categorical variables of the dataset.
     '''
     df = add_counters_for_all_features(df, features, counters, time_windows)
     df = pd.get_dummies(df, columns=['student_id', 'lesson_id', 'exercise_code', 'learning_object', 'exercise_code_level'])
     return df
 
-def add_counters_for_all_features(df, features, counters, time_windows):
+def add_counters_for_all_features(df: pd.DataFrame, features: tuple, counters: tuple, time_windows: tuple):
     '''
     Adds a column for all given features, all given counters of a student and all given time windows.
     '''
     for feature in features:
-        df = add_feature_counters(df, feature, counters, time_windows)
+        df = add_feature_counters_in_all_time_windows(df, feature, counters, time_windows)
     return df
 
-def add_feature_counters(df, feature, counters, time_windows):
+def add_feature_counters_in_all_time_windows(df: pd.DataFrame, feature: str, counters: tuple, time_windows: tuple):
```

Comment:
I would name it `...for_each_time_window` because `all_time` can be misleading i think